### DISCUSSION TITLE
Bug? Using Fief behind dynamic IP looses admin session

### DISCUSSION THREAD

**Discussion by ClaudioBsh** [fief-dev/fief#328] [General]
I test fief on a Raspberry in my home network.

My internet provider sometimes cancel the connection, after reconnect my home network is getting a new dynamic public ip.

I have a domain registered at Cloudflare DNS.

With a bashscript i check all 2mins for a new public ip, and if the ip changed the script is updating the ip at Cloudflare so my domain will be reachable again.

So far so good.

But i had open an fief admin session during provider canceld current connection and the admin session is broken, still after reconnect with new public ip.

So far ok, but i never got it running. 

I saw error messages in the fief server log (did not save it) and got http code 400 and 500 errors. I did restart fief server, tried to login again to fief.mydomain/admin - but same problems ocurs.

Then i did restart the fief worker, does also not help. 

Then i stopped both, then started both again, now i can login to the admin page.

So i wonder, what could be or better what was/is the reason for this problem?

Any ideas?

**Comment by fief-bailiff** [comment]
Hail, @ClaudioBsh 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to star our repository ⭐️

![star-fief](https://user-images.githubusercontent.com/1144727/192523552-dc2c31fa-ae94-42fa-9ab4-7d0ba51bb7b0.png)

Farewell!

**Reply by ClaudioBsh** [reply]
Ok, after some tests i cannot confirm that this situation is reproduceable.

So for me it seems there was another problem, not the public ip change was the reason. 

If the problem will occur again, i will post the log here.

P.S. I use fief together with Redis - just to be have this in mind....could this results in trouble when connection is getting lost and the public ip did change?

**Comment by frankie567** [comment]
The admin session is maintained by a browser cookie, so theoretically, if your domain stays the same (even though the underlying IP changes), the browser should do its work and forward the cookie.

Maybe there was another problem meanwhile, which was not related to the dynamic IP thing. Keep us tuned!
